database/src/main/java/com/firebase/ui/database/FirebaseIndexRecyclerAdapter.java

```diff
@@ -59,8 +59,11 @@
  * </pre>
  *
  * @param <T>  The Java class that maps to the type of objects stored in the Firebase location.
- * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
+ * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each
+ *             object.
+ * @deprecated use {@link com.firebase.ui.database.adapter.FirebaseIndexRecyclerAdapter} instead
  */
+@Deprecated
```

Comment:
Let's take this as a concrete example of the deprecation objection.  Here are the differences I see in the new class vs the old class:

- A change in the order of the constructor arguments (`viewHolderClass` switched with `modelLayout`)
- A change in the package name
- `dataRef` has become a `DatabaseReference` where it was a `Query`

The first change (param ordering) seems to just be random.  The package name change is nice, but certainly the kind of thing we can live with unchanged.  And the third change could be done by simply deprecating the constructor that accepts `Query` and pointing people to a new overloaded constructor that accepts a `DatabaseReference`

So couldn't we avoid deprecating this class?